From 953a7dd92c4facdf48c390cb9b3f1672a971ac05 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Sun, 05 Jun 2022 15:56:53 +0300
Subject: [PATCH] Revert "[SW-82716] common: add fw upd routine for i2c"

Revert rom recovery implementation for secured firmware update via
inband and OOB.

This reverts commit f70d53cd20b240aaedb03d61ed8c8aa43583f556.

Change-Id: I6f499b1ed882a7021c69e06b1f00bd6e9942a61e
---

diff --git a/arch/gaudi3/arc/recovery.c b/arch/gaudi3/arc/recovery.c
index 66046c8..c765289 100644
--- a/arch/gaudi3/arc/recovery.c
+++ b/arch/gaudi3/arc/recovery.c
@@ -40,16 +40,6 @@
 		.teardown = noop,
 		.run_noreturn = module_fwupd_run_noret,
 	},
-	{	/* I2C FW UPD module */
-		.setup = module_i2c_fwupd_setup,
-		.check = module_i2c_fwupd_check,
-		.disable = noop,
-		.run = module_i2c_fwupd_run,
-		.enable = noop,
-		.run_fast = module_i2c_fwupd_run_fast,
-		.teardown = noop,
-		.run_noreturn = module_fwupd_run_noret,
-	},
 };
 
 static enum dma_dest_type arcboot_rom_get_dma_dest_type(uint64_t addr)
diff --git a/common/cmd_module.h b/common/cmd_module.h
index 9add463..61268ad 100644
--- a/common/cmd_module.h
+++ b/common/cmd_module.h
@@ -84,14 +84,5 @@
 void module_fwupd_pcie_disable(bool is_pcie_disabled);
 bool module_fwupd_is_disabled(void);
 
-/* I2C FWUPD */
-void module_i2c_fwupd_setup(void);
-enum cmd_check_res module_i2c_fwupd_check(void);
-void module_i2c_fwupd_run_fast(void);
-void module_i2c_fwupd_run(void);
-bool i2c_fwupd_owned(uint32_t offset);
-uint32_t i2c_fwupd_readl(uint32_t offset);
-void i2c_fwupd_writel(uint32_t offset, uint32_t val);
-
 void module_handle_cmds(struct cmd_module_ops cmd_ops[], int size);
 #endif /* _HL_CMD_MODULE_H_ */
diff --git a/common/module_fwupd.c b/common/module_fwupd.c
index fc02b2a..3730c19 100644
--- a/common/module_fwupd.c
+++ b/common/module_fwupd.c
@@ -58,40 +58,11 @@
 };
 
 static struct fwupd_state state;
-static struct fwupd_state i2c_state;
 static struct fwupd_resp_prop resp_props[RESP_PROP_NUM];
-
-
-uint32_t i2c_fwupd_cmd_readl(void)
-{
-	return i2c_state.cmd_reg;
-}
-
-void i2c_fwupd_sts_writel(uint32_t val)
-{
-	i2c_state.sts_reg = val;
-}
 
 uint32_t fwupd_cmd_readl(void)
 {
 	return regmap_readl(global_conf.upd_cmd_map);
-}
-
-bool i2c_fwupd_owned(uint32_t offset)
-{
-	return offset == global_conf.upd_cmd_map || offset == global_conf.upd_sts_map;
-}
-
-/* This function is for reading status by other modules */
-uint32_t i2c_fwupd_readl(__unused uint32_t offset)
-{
-	return i2c_state.sts_reg;
-}
-
-/* This function is for writing cmd by other modules */
-void i2c_fwupd_writel(__unused uint32_t offset, uint32_t val)
-{
-	i2c_state.cmd_reg = val;
 }
 
 void fwupd_sts_writel(uint32_t val)
@@ -281,16 +252,6 @@
 	module_fwupd_send_status(&state, HL_UPD_STS_RST_OK, NULL);
 }
 
-void module_i2c_fwupd_setup(void)
-{
-	i2c_state.cmd_readl = i2c_fwupd_cmd_readl;
-	i2c_state.sts_writel = i2c_fwupd_sts_writel;
-	i2c_state.use_security = false;
-	i2c_state.synced_no_return = true;
-	i2c_state.mregion = MEM_I2C;
-	module_fwupd_send_status(&i2c_state, HL_UPD_STS_RST_OK, NULL);
-}
-
 /* NOOP is a sync command, so here we decide what execution type is needed */
 static enum cmd_check_res module_fwupd_check_noop(struct fwupd_state *state)
 {
@@ -361,11 +322,6 @@
 enum cmd_check_res module_fwupd_check(void)
 {
 	return __module_fwupd_check(&state);
-}
-
-enum cmd_check_res module_i2c_fwupd_check(void)
-{
-	return __module_fwupd_check(&i2c_state);
 }
 
 static void module_fwupd_state_shift(struct fwupd_state *state)
@@ -505,11 +461,6 @@
 	__module_fwupd_run_fast(&state);
 }
 
-void module_i2c_fwupd_run_fast(void)
-{
-	__module_fwupd_run_fast(&i2c_state);
-}
-
 static int module_fwupd_load_type_run(struct fwupd_state *state, void *fit)
 {
 	int err;
@@ -618,11 +569,6 @@
 void module_fwupd_run(void)
 {
 	__module_fwupd_run(&state);
-}
-
-void module_i2c_fwupd_run(void)
-{
-	__module_fwupd_run(&i2c_state);
 }
 
 void module_fwupd_run_noret(void)
